src/graph.cpp

```diff
@@ -229,4 +236,61 @@ std::vector<Vertex *> Graph::findEndOfLines(const int stationId) const {
     }
 
     return eol_stations;
-}
\ No newline at end of file
+}
+
+/**
+ * DFS traversal variation that sets the visited attribute to true of the nodes the DFS traverses to
+ * Time Complexity: O(|V|+|E|)
+ * @param source - Vertex where the DFS starts
+ */
+void Graph::visitedDFS(Vertex *source) {
+    source->setVisited(true);
+    for (Edge *e : source->getAdj()){
+        if (!e->getDest()->isVisited()){
+            visitedDFS(e->getDest());
+        }
+    }
+}
+
+/**
+ * Finds the pairs of stations with max Max-Flow
+ * Time Complexity: O(V^2*(VE^2)) //TODO someone check if V^2 is the correct ammount of times EdmondsKarp is called
+ * @return
+ */
+std::pair<std::list<std::pair<Vertex *,Vertex *>>,unsigned int> Graph::calculateNetworkMaxFlow() {
+    unsigned int max = -1;
+    std::list<std::pair<Vertex *,Vertex *>> stationList;
+    for (auto itV1 = vertexSet.begin(); itV1 < vertexSet.end(); itV1++){
+        for (auto itV2 = itV1; itV2 < vertexSet.end(); itV2++){
+            Vertex * v1 = *itV1++;
```

Comment:
What is the purpose of incrementing v1 here if you already do it in the cycle? Also, shouldn't v2 start on v1 +1 as to avoid getting the max flow between a vertex and itself?

Reply:
Oops, that was just a typo, consider it fixed